SWIG/volatilities.i

```diff
@@ -661,12 +661,12 @@ class SwaptionVolCube1 : public SwaptionVolatilityCube {
     Matrix marketVolCube() const;
     Matrix volCubeAtmCalibrated() const;
     %extend {
-        ext::shared_ptr<SabrSmileSection> smileSection(Time optionTime, Time swapLength, bool extr = false) {
-            SwaptionVolatilityStructure* base = dynamic_cast<SwaptionVolatilityStructure*>($self);
+        ext::shared_ptr<SabrSmileSection> smileSection(Time optionTime, Time swapLength, bool extr = false) const {
+            SwaptionVolatilityStructure* base = dynamic_cast<SwaptionVolatilityStructure*>(const_cast<SwaptionVolCube1*>($self));
```

Comment:
@lballabio I had to include a `const_cast` here to be able to make the method `const` itself. Is this correct?

Reply:
I think you can use `dynamic_cast<const SwaptionVolatilityStructure*>($self)` instead.

Reply:
You are right. Using `shared_ptr` all the time I forgot about the good old pointer arithmetic.